### ISSUE TITLE
Set-up Hot Jar tracking code in Connect

### ISSUE THREAD

**Issue by acshields** [appirio-tech/connect-app#2475]
@vikasrohit - To monitor current registration form performance vs. the performance of the new registration form, I would like to leverage Hot Jar.  In order to do this a tracking code needs to be installed on every page in Connect (I am asking for every page because I would like to leverage Hot Jar for other reporting within Connect as well so it would be good to accomplish this at one time).

**Here is the tracking code:**

<script>
    (function(h,o,t,j,a,r){
        h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};
        h._hjSettings={hjid:133592,hjsv:6};
        a=o.getElementsByTagName('head')[0];
        r=o.createElement('script');r.async=1;
        r.src=t+h._hjSettings.hjid+j+h._hjSettings.hjsv;
        a.appendChild(r);
    })(window,document,'https://static.hotjar.com/c/hotjar-','.js?sv=');
</script>

Here is an overview of the process.  You can also review installation instruction via this link: https://help.hotjar.com/hc/en-us/articles/115009499088-The-Hotjar-Tracking-Code

![screen shot 2018-09-06 at 9 51 33 am](https://user-images.githubusercontent.com/41118149/45173119-19d29800-b1bc-11e8-8274-ab80ebd7ea03.png)

![screen shot 2018-09-06 at 9 51 46 am](https://user-images.githubusercontent.com/41118149/45173123-1d661f00-b1bc-11e8-9a0a-72ca4e2a4d21.png)

Aha! Link: https://topcoder.aha.io/features/TCCONNECT-448

**Comment by vikasrohit** [comment]
@acshields I am not sure what extra details hotjar provides to us, but we already have many tracking and analytics enabled on connect via segment.
<img width="239" alt="screen shot 2018-09-10 at 5 30 41 pm" src="https://user-images.githubusercontent.com/2417632/45296314-4a4d6580-b51f-11e8-9738-7271fb18671a.png">

That being said, do we have accounts credentials for hotjar? I might need those to validate the integration.

I would do this tomorrow.

**Comment by ajefts** [comment]
@vikasrohit We can actually add hotjar via segment, BUT...we need to switch the way Connect is setup in segment first.  If you can update the segment include to the below then we can handle the rest from segment.

```
<script>
  !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t,e){var n=document.createElement("script");n.type="text/javascript";n.async=!0;n.src="https://cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(n,a);analytics._loadOptions=e};analytics.SNIPPET_VERSION="4.1.0";
  analytics.load("o9nnBWWDraSZHiuruzcSfaNVJFphGq5G");
  analytics.page();
  }}();
</script>

```

**Comment by vikasrohit** [comment]
@ajefts Does this snippet only changes the key of segment or does it change the type of the source in the segment?

**Comment by vikasrohit** [comment]
@ajefts just to confirm, its only about changing the segment source?

**Comment by vikasrohit** [comment]
@acshields @ajefts I have configured the production instance of Connect to use the `Connect 2` segment source as suggested by Tony. And it started to pick the events. However, it was not getting the page events for internal route changes which are essentials for SPA and it used to work in past but I don't know when it stopped working (I guess, when last time we upgraded react and react-router, it broke). I have applied a patch to the production so that now it receives the internal route change events as page events in Segment so we can track all pages opened by a user.

Seems like hotjar is also configured by Tony with `Connect 2` source, so most probably we should be getting the events in hotjar as well. I could not verify that without credentials of hotjar.
